diag_manager/fms_send_data_statfun.F90

```diff
@@ -0,0 +1,1139 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @defgroup fms_send_data_statfun_mod fms_send_data_statfun_mod
+!> @ingroup diag_manager
+!> @brief fms_send_data_statfun_mod contains functions used by the
+!! diag_managers send_data routines tto prepare the filed/data for output.
+!!  The functions include agregation, min and max, etc.
+!!
+!! <TT>fms_send_data_stattfun_mod</TT>  defines functions used by the
+!! diag_managers send_data routines tto prepare the filed/data for output.
+!!  The functions include agregation, min and max, etc.
+!!
+!> @file
+!> @brief File for @ref fms_send_data_statfun_mod
+!> @addtogroup fms_diag_dlinked_list_mod
+!> @{
+MODULE fms_send_data_statfun_mod
+
+  use platform_mod
+   USE mpp_mod, ONLY: mpp_pe, mpp_root_pe
+
+   USE fms_mod, ONLY: error_mesg, FATAL, WARNING, NOTE, stdout, stdlog, write_version_number,&
+   & fms_error_handler
+   USE diag_data_mod, ONLY:  input_fields, output_fields, debug_diag_manager
```

Comment:
This is old code.  Are you planning on integrating the diag_object?

Reply:
Yes. See item below (https://github.com/NOAA-GFDL/FMS/pull/979#discussion_r878136039)